scripts/o3de/o3de/repo_properties.py

```diff
@@ -92,17 +222,47 @@ def create_remote_object_archive(src_data_path: pathlib.Path,
         FileNotFoundError: If the json_data_path does not exist.
         ValueError: If the json_data_path is not a dict.
     """
-    if download_prefix is None:
-        logger.error('The --download-prefix argument must be provided. A url prefix for a file attached to a Github release might look like this:'
-                     '-dp https://github.com/o3de/o3de-extras/releases/download/2305.0')
-        return {}
     zip_path = releases_path / archive_filename
     # check if a object.zip folder already exist in the path - ask user if they want to overwrite the current zip
     if not force and zip_path.exists():
         logger.error(f'{zip_path} already exists.  Use --force command to overwrite the existing zip or '
                      'provide a new location to save your archive.')
         return {}
-    else:
+    if upload_github_release_tag is None and download_prefix is None:
+        logger.error('You need to provide a --upload-github-release-tag if you want to upload to github.\n'
+                     'If you want to use other version control systems you must provide --download-prefix argument\n'
+                     'A url prefix for a file attached to a Github release might look like this:'
+                     '-dp https://github.com/o3de/o3de-extras/releases/download/2305.0')
+        return {}
+    if upload_github_release_tag and download_prefix is None:
+        
+        if _is_github_link(repo_uri):
+        # Parse the URL to extract owner and repository name
+            owner_repo = repo_uri.split("github.com/")[1].rstrip('.git').split('/')
+            owner = owner_repo[0]
+            repo = owner_repo[1]
```

Comment:
here I recommend we get a GitProvider instead so it isn't limited to GitHub like so:
```python
if upload_git_release_tag and download_prefix is None:
    git_provider = github_utils.get_github_provider(repo_uri)
    if git_provider:
        git_provider.upload_release(...)
    else:
        logger.error(f"Failed to determine Git provider from {repo_uri}")